src/components/ha-selector/ha-selector-select.ts

```diff
@@ -60,36 +142,121 @@ export class HaSelectSelector extends LitElement {
         @closed=${stopPropagation}
         @selected=${this._valueChanged}
       >
-        ${this.selector.select.options.map((item: string | SelectOption) => {
-          const value = typeof item === "object" ? item.value : item;
-          const label = typeof item === "object" ? item.label : item;
-
-          return html`<mwc-list-item .value=${value}>${label}</mwc-list-item>`;
-        })}
+        ${options.map(
+          (item: SelectOption) => html`
+            <mwc-list-item .value=${item.value}>${item.label}</mwc-list-item>
+          `
+        )}
       </ha-select>
     `;
   }
 
+  private get _mode(): "list" | "dropdown" {
+    return this.selector.select.mode || this.selector.select.options.length > 6
+      ? "list"
+      : "dropdown";
+  }
+
   private _valueChanged(ev) {
     ev.stopPropagation();
-    if (this.disabled || !ev.target.value) {
+    const value = ev.detail?.value || ev.target.value;
+    if (this.disabled || !value) {
       return;
     }
     fireEvent(this, "value-changed", {
-      value: ev.target.value,
+      value: value,
     });
   }
 
-  static get styles(): CSSResultGroup {
-    return css`
-      ha-select {
-        width: 100%;
+  private _checkboxChanged(ev) {
+    ev.stopPropagation();
+    if (this.disabled) {
+      return;
+    }
+
+    let newValue: string[];
+    const value: string = ev.target.value;
+    const checked = ev.target.checked;
+
+    if (checked) {
+      if (!this.value) {
+        newValue = [value];
+      } else if (this.value.includes(value)) {
+        return;
+      } else {
+        newValue = [...this.value, value];
       }
-      mwc-formfield {
-        display: block;
+    } else {
+      if (!this.value?.includes(value)) {
+        return;
       }
-    `;
+      newValue = (this.value as string[]).filter((v) => v !== value);
+    }
+
+    fireEvent(this, "value-changed", {
+      value: newValue,
+    });
   }
+
+  private _removeItem(ev): void {
+    (this.value as string[])!.splice(ev.target.idx, 1);
+
+    fireEvent(this, "value-changed", {
+      value: this.value,
+    });
+    this.requestUpdate();
+  }
+
+  private _comboBoxValueChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+    const newValue = ev.detail.value;
+
+    if (this.disabled || newValue === "") {
+      return;
+    }
+
+    if (!this.selector.select.multiple) {
+      fireEvent(this, "value-changed", {
+        value: newValue,
+      });
+      return;
+    }
+
+    // Trying to reset the value... Not working currently
+    this._filter = "";
+    this.comboBox.value = "";
```

Comment:
I know it's just experimenting to get it working, but we should never manipulate values that are being set by Lit.